sdk/cosmos/azure-cosmos-spark_3-1_2-12/src/test/scala/com/azure/cosmos/spark/SparkE2EConfigResolutionITest.scala

```diff
@@ -97,6 +97,32 @@ class SparkE2EConfigResolutionITest extends IntegrationSpec with CosmosClient wi
     spark.close()
   }
 
+  it should "validate config names with 'spark.cosmos.' prefix" in {
+    val userConfig = Map(
+      "spark.cosmos.accountEndpoint" -> "https://boson-test.documents.azure.com:443/",
+      "spark.cosmos.accountKey" -> "xyz",
+      "spark.cosmos.someTypo" -> "xyz",
+    )
+
+    try {
+      CosmosConfig.getEffectiveConfig(None, None, userConfig)
+      fail("Should throw on invalid config names")
+    } catch {
+      case e: Exception => e.getMessage shouldEqual
```

Comment:
Would it be the same throw/type if the invalid config is used to execute a query for example?

Reply:
Yes - getEffectiveConfig is called in all scenarios (and I accidentally tested how it works on a query because we had an invalid setting in one of our unit tests for query :-))